geo/geo.go

```diff
@@ -39,7 +41,21 @@ func (l *Location) Empty() bool {
 
 func (l *Location) Match(location *Location) bool {
 	if len(l.IP) > 0 {
-		if location.IP == l.IP {
+		matched, err := regexp.MatchString(`/\d{1,2}`, l.IP)
+		if err != nil {
+			log.Println(err)
+			return false
+		}
+		if !matched {
+			l.IP += "/32"
+		}
+		_, subnet, err := net.ParseCIDR(l.IP)
```

Comment:
I would suggest we add a field `cidr` in `Location` and cache the subnet so we only parse it once